src/materials/ADComputeISoilStress.C

```diff
@@ -0,0 +1,642 @@
+                                       "The over consolidation ratio of the soil "
+                                       "layers. Required for Darandeli backbone curve.");
+    params.addParam<std::vector<Real>>(
+        "plasticity_index",
+        "The plasticity index of the soil layers. Required for Darandeli backbone curve.");
+    // params required for GQ/H backbone curve: soil_type = "gqh"
+    params.addParam<std::vector<Real>>("theta_1",
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("theta_2",
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("theta_3",
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("theta_4",
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("theta_5",
+                                       "The curve fit coefficient for "
+                                       "GQ/H model"
+                                       "for each soil layer.");
+    params.addParam<std::vector<Real>>("taumax",
+                                       "The ultimate shear strength of "
+                                       "the soil layers. Required for "
+                                       "GQ/H model");
+    // params required for thin_layer contact model soil_type = "thin_layer"
+    params.addParam<std::vector<Real>>("friction_coefficient",
+                                       "Friction coefficients of the thin layers.");
+    params.addParam<std::vector<Real>>("hardening_ratio",
+                                       "Post-yield hardening ratios of the layers."););
+
+template <ComputeStage compute_stage>
+ADComputeISoilStress<compute_stage>::ADComputeISoilStress(const InputParameters & parameters)
+  : ADComputeFiniteStrainElasticStress<compute_stage>(parameters),
+    _base_models(),
+    _stress_model(),
+    _stress_model_old(),
+    _yield_stress(), // *** YIELD STRAIN NOT YIELD STRESS ***
+    _youngs(),
+    _soil_layer_variable(coupledValue("layer_variable")),
+    _layer_ids(getParam<std::vector<unsigned int>>("layer_ids")),
+    _wave_speed_calculation(getParam<bool>("wave_speed_calculation")),
+    _poissons_ratio(getParam<std::vector<Real>>("poissons_ratio")),
+    _density(_wave_speed_calculation ? &getMaterialProperty<Real>("density") : NULL),
```

Comment:
Do we know that `density` is always going to be independent of the our non-linear degrees of freedom?

Reply:
`nullptr` is preferable to `NULL`

Reply:
Yes, the density is a material property that is provided by the user and it is independent of nonlinear variables for all the problems we solve using MASTODON. I will change NULL to nullptr